src/Microsoft.ServiceFabric.Actors.Migration/Extensions/KvsActorStateProviderExtensionHelper.cs

```diff
@@ -0,0 +1,188 @@
+                                {
+                                    var keyValuePair = MakeKeyValuePair(enumerator.Current);
+                                    var currentSequenceNumber = keyValuePair.Version;
+
+                                    pairs.Add(keyValuePair);
+                                    enumerationKeyCount++;
+
+                                    hasData = enumerator.MoveNext();
+
+                                    if (hasData)
+                                    {
+                                        var nextKeyValuePair = enumerator.Current;
+                                        var nextKeySequenceNumber = nextKeyValuePair.Metadata.SequenceNumber;
+                                        sequenceNumberFullyDrained = !(nextKeySequenceNumber == currentSequenceNumber);
+                                    }
+                                }
+
+                                var keyvaluepairserializer = new DataContractSerializer(typeof(List<KeyValuePair>));
+                                using var memoryStream = new MemoryStream();
+                                var binaryWriter = XmlDictionaryWriter.CreateBinaryWriter(memoryStream);
+                                keyvaluepairserializer.WriteObject(binaryWriter, pairs);
+                                binaryWriter.Flush();
+
+                                var byteArray = memoryStream.ToArray();
+
+                                ActorTrace.Source.WriteInfo("KvsActorStateProviderExtensionHelper", $"ByteArray: {byteArray} ArrayLength: {byteArray.Length} StreamLength: {memoryStream.Length}");
+
+                                // Set the content type
+                                response.ContentType = "application/xml; charset=utf-8";
+                                await response.Body.WriteAsync(byteArray, 0, byteArray.Length);
+                                await response.Body.FlushAsync();
+                            }
+                        }
+                    }
+                    catch (Exception e)
+                    {
+                        ActorTrace.Source.WriteError("KvsActorStateProviderExtensionHelper", $"{e.Message} {e.StackTrace}");
+                        if (e.InnerException != null)
+                        {
+                            ActorTrace.Source.WriteError("KvsActorStateProviderExtensionHelper", $"{e.InnerException.Message} {e.InnerException.StackTrace}/n");
+                        }
+                    }
+                },
+                "EnumerateAsync",
+                cancellationToken);
+        }
+#endif
+
+        internal static bool TryAbortExistingTransactionsAndRejectWrites(this KvsActorStateProvider stateProvider)
+        {
+            return stateProvider.GetStoreReplica().TryAbortExistingTransactionsAndRejectWrites();
+        }
+
+        internal static async Task SaveKvsRejectWriteStatusAsync(this KvsActorStateProvider stateProvider, bool ready)
```

Comment:
Can you think of a better api name? KVSSP.RejectWritesAsync() and KVSSP.ResumeWritesAsync() sounds better?